hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/web/resources/DeleteSkipTrashParam.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdfs.web.resources;
+
+/**
+ * SkipTrash param to be used by DELETE query.
+ */
+public class DeleteSkipTrashParam extends BooleanParam {
+
+  public static final String NAME = "skiptrash";
+  public static final String DEFAULT = FALSE;
```

Comment:
We should set the WebHDFS delete skiptrash default to `true` imo in order to behave the same with previous versions. This can be an unexpected behavior change when a cluster/client is upgraded. This means if a WebHDFS client doesn't want skiptrash it should explicitly state `skiptrash=false`.

I can understand `skiptrash=false` is the default for CLI, but for WebHDFS it can be seen as implicit `skiptrash=true` since the beginning (and is the norm for WebHDFS users).

Thoughts?

Reply:
Thanks for the suggestion. In fact, there was similar discussion on Jira as well and it seems so far the consensus was to keep it `false` by default. Because of which, this will be incompatible change w.r.t default behaviour of DELETE API. 
Hence, the decision was to mark Jira Incompatible change and we can still go ahead with this new behaviour starting 3.3.1/3.4.0 release.
 
However, I am fine changing this to `true` as well if that's where majority would like to go with.

FYI @jojochuang

Reply:
Alright, if the jira has the incompatible label I'm fine with skiptrash=false default. :)

@jojochuang

Reply:
Ah I just noticed target version includes 3.3.1, backporting to 3.3.x might be a problem if this is an incompatible change.

According to the [compatibility guideline](https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-common/Compatibility.html#REST_APIs):
> Each API has an API-specific version number. Any incompatible changes MUST increment the API version number.

How about this:
- Change `skiptrash=true` default to be compatible with WebHDFS v1, backport this to 3.3.1
- Set `skiptrash=false` in a separate jira for 3.4.0, which will be an incompatible change

Or:
- Increment WebHDFS REST API version to v2 which has `skiptrash=false` as default for DELETE

Reply:
I understand your concerns @smengcl.
@jojochuang's [comment](https://issues.apache.org/jira/browse/HDFS-15982?focusedCommentId=17331521&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17331521) from Jira that you might be interested to look at:
```
This is a big incompatible change. If we think this should be part of 3.4.0, risking our compatibility guarantee (which I think makes sense, given how many times I was involved in accidental data deletion), I think it can be part of 3.3.1. We traditionally regard 3.3.0 as non-production ready, so making an incompat change in 3.3.1 probably is justifiable.
```

Reply:
👌. Let's include the incompatible change note in 3.3.1 and 3.4.0 release notes.